pkg/ds/daemon_set.go

```diff
@@ -450,51 +534,17 @@ func (ds *daemonSet) reportEligible() error {
 	return nil
 }
 
-// Watch for changes to nodes and sends update and delete signals
-func (ds *daemonSet) handleNodeChanges(changes *labels.LabeledChanges) error {
-	if len(changes.Updated) > 0 {
-		ds.logger.NoFields().Infof("Received node change signal")
-		err := ds.removePods()
-		if err != nil {
-			return util.Errorf("Unable to remove pods from intent tree: %v", err)
-		}
-		err = ds.addPods()
-		if err != nil {
-			return util.Errorf("Unable to add pods to intent tree: %v", err)
-		}
-		return nil
-	}
-
-	if len(changes.Created) > 0 {
-		ds.logger.NoFields().Infof("Received node create signal")
-		err := ds.addPods()
-		if err != nil {
-			return util.Errorf("Unable to add pods to intent tree: %v", err)
-		}
-	}
-
-	if len(changes.Deleted) > 0 {
-		ds.logger.NoFields().Infof("Received node delete signal")
-		err := ds.removePods()
-		if err != nil {
-			return util.Errorf("Unable to remove pods from intent tree: %v", err)
-		}
-	}
-
-	return nil
-}
-
 // addPods schedules pods for all unscheduled nodes selected by ds.nodeSelector
-func (ds *daemonSet) addPods() error {
+func (ds *daemonSet) addPods() ([]types.NodeName, error) {
```

Comment:
Is it too much to ask for a rename of this function? It has no side-effects, it's just doing computation.

Reply:
not at all!